From f21139e60b515d14fecb98344913a0cc35a8e2bc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Uwe=20Kleine-K=C3=B6nig=20=28The=20Capable=20Hub=29?=
 <u.kleine-koenig@baylibre.com>
Date: Fri, 17 Jul 2026 16:50:49 +0200
Subject: [PATCH 09284/15333] misc: genwqe: Drop unused assignment of
 pci_device_id driver data
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The driver explicitly sets the .driver_data member of struct
pci_device_id to zero without relying on that value. Drop this unused
assignment.

While touching this array, also convert it to use one of the PCI_DEVICE
macros that is usually used for these and drop the explicit 0 in the
list terminator. This patch doesn't modify the compiled array.

Signed-off-by: Uwe Kleine-König (The Capable Hub) <u.kleine-koenig@baylibre.com>
Link: https://patch.msgid.link/0756a50095f65ada4b82a65a4f827bcd5f0e5d84.1784299069.git.u.kleine-koenig@baylibre.com
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/misc/genwqe/card_base.c | 88 +++++++++++++--------------------
 1 file changed, 35 insertions(+), 53 deletions(-)

diff --git a/drivers/misc/genwqe/card_base.c b/drivers/misc/genwqe/card_base.c
index 86bfa82723ff..e602eb3c51fa 100644
--- a/drivers/misc/genwqe/card_base.c
+++ b/drivers/misc/genwqe/card_base.c
@@ -48,60 +48,42 @@ static struct genwqe_dev *genwqe_devices[GENWQE_CARD_NO_MAX];
 
 /* PCI structure for identifying device by PCI vendor and device ID */
 static const struct pci_device_id genwqe_device_table[] = {
-	{ .vendor      = PCI_VENDOR_ID_IBM,
-	  .device      = PCI_DEVICE_GENWQE,
-	  .subvendor   = PCI_SUBVENDOR_ID_IBM,
-	  .subdevice   = PCI_SUBSYSTEM_ID_GENWQE5,
-	  .class       = (PCI_CLASSCODE_GENWQE5 << 8),
-	  .class_mask  = ~0,
-	  .driver_data = 0 },
-
-	/* Initial SR-IOV bring-up image */
-	{ .vendor      = PCI_VENDOR_ID_IBM,
-	  .device      = PCI_DEVICE_GENWQE,
-	  .subvendor   = PCI_SUBVENDOR_ID_IBM_SRIOV,
-	  .subdevice   = PCI_SUBSYSTEM_ID_GENWQE5_SRIOV,
-	  .class       = (PCI_CLASSCODE_GENWQE5_SRIOV << 8),
-	  .class_mask  = ~0,
-	  .driver_data = 0 },
-
-	{ .vendor      = PCI_VENDOR_ID_IBM,  /* VF Vendor ID */
-	  .device      = 0x0000,  /* VF Device ID */
-	  .subvendor   = PCI_SUBVENDOR_ID_IBM_SRIOV,
-	  .subdevice   = PCI_SUBSYSTEM_ID_GENWQE5_SRIOV,
-	  .class       = (PCI_CLASSCODE_GENWQE5_SRIOV << 8),
-	  .class_mask  = ~0,
-	  .driver_data = 0 },
-
-	/* Fixed up image */
-	{ .vendor      = PCI_VENDOR_ID_IBM,
-	  .device      = PCI_DEVICE_GENWQE,
-	  .subvendor   = PCI_SUBVENDOR_ID_IBM_SRIOV,
-	  .subdevice   = PCI_SUBSYSTEM_ID_GENWQE5,
-	  .class       = (PCI_CLASSCODE_GENWQE5_SRIOV << 8),
-	  .class_mask  = ~0,
-	  .driver_data = 0 },
-
-	{ .vendor      = PCI_VENDOR_ID_IBM,  /* VF Vendor ID */
-	  .device      = 0x0000,  /* VF Device ID */
-	  .subvendor   = PCI_SUBVENDOR_ID_IBM_SRIOV,
-	  .subdevice   = PCI_SUBSYSTEM_ID_GENWQE5,
-	  .class       = (PCI_CLASSCODE_GENWQE5_SRIOV << 8),
-	  .class_mask  = ~0,
-	  .driver_data = 0 },
-
-	/* Even one more ... */
-	{ .vendor      = PCI_VENDOR_ID_IBM,
-	  .device      = PCI_DEVICE_GENWQE,
-	  .subvendor   = PCI_SUBVENDOR_ID_IBM,
-	  .subdevice   = PCI_SUBSYSTEM_ID_GENWQE5_NEW,
-	  .class       = (PCI_CLASSCODE_GENWQE5 << 8),
-	  .class_mask  = ~0,
-	  .driver_data = 0 },
-
-	{ 0, }			/* 0 terminated list. */
+	{
+		PCI_DEVICE_SUB(PCI_VENDOR_ID_IBM, PCI_DEVICE_GENWQE,
+			       PCI_SUBVENDOR_ID_IBM, PCI_SUBSYSTEM_ID_GENWQE5),
+		.class       = PCI_CLASSCODE_GENWQE5 << 8,
+		.class_mask  = ~0,
+	}, {
+		/* Initial SR-IOV bring-up image */
+		PCI_DEVICE_SUB(PCI_VENDOR_ID_IBM, PCI_VENDOR_ID_IBM, PCI_DEVICE_GENWQE,
+			       PCI_SUBVENDOR_ID_IBM_SRIOV),
+		.class       = PCI_CLASSCODE_GENWQE5_SRIOV << 8,
+		.class_mask  = ~0,
+	}, {
+		PCI_DEVICE_SUB(PCI_VENDOR_ID_IBM, 0x0000,
+			       PCI_SUBVENDOR_ID_IBM_SRIOV, PCI_SUBSYSTEM_ID_GENWQE5_SRIOV),
+		.class       = PCI_CLASSCODE_GENWQE5_SRIOV << 8,
+		.class_mask  = ~0,
+	}, {
+		/* Fixed up image */
+		PCI_DEVICE_SUB(PCI_VENDOR_ID_IBM, PCI_DEVICE_GENWQE,
+			       PCI_SUBVENDOR_ID_IBM_SRIOV, PCI_SUBSYSTEM_ID_GENWQE5),
+		.class       = PCI_CLASSCODE_GENWQE5_SRIOV << 8,
+		.class_mask  = ~0,
+	}, {
+		PCI_DEVICE_SUB(PCI_VENDOR_ID_IBM, 0x0000,
+			       PCI_SUBVENDOR_ID_IBM_SRIOV, PCI_SUBSYSTEM_ID_GENWQE5),
+		.class       = PCI_CLASSCODE_GENWQE5_SRIOV << 8,
+		.class_mask  = ~0,
+	}, {
+		/* Even one more ... */
+		PCI_DEVICE_SUB(PCI_VENDOR_ID_IBM, PCI_DEVICE_GENWQE,
+			       PCI_SUBVENDOR_ID_IBM, PCI_SUBSYSTEM_ID_GENWQE5_NEW),
+		.class       = (PCI_CLASSCODE_GENWQE5 << 8),
+		.class_mask  = ~0,
+	},
+	{ }			/* 0 terminated list. */
 };
-
 MODULE_DEVICE_TABLE(pci, genwqe_device_table);
 
 /**
-- 
2.55.0

